### PR TITLE
Handle git errors while cloning

### LINKED ISSUE #57: [Bug] Installer should Respect transport protocol (git settings)
#### Current Behavior
Currently the installer (at least the ps1 script for windows) tries to install scoop by cloning the repo **only** via HTTPS:
```ps1
# ...
$SCOOP_PACKAGE_GIT_REPO = "https://github.com/ScoopInstaller/Scoop.git"
# ...
git clone -q $SCOOP_PACKAGE_GIT_REPO $SCOOP_APP_DIR
```
If the HTTPS transport protocol is disabled in the git configuration of a user/system, the installer breaks.

#### Expected Behavior
The installer should try to fallback to SSH.

#### Additional context/output
This is my installer output:
```log
PS> irm get.scoop.sh | iex
Initializing...
Downloading ...
fatal: transport 'https' not allowed
fatal: transport 'https' not allowed
Creating shim...
Resolve-Path : Cannot find path 'C:\Users\u6\scoop\apps\scoop\current\bin\scoop.ps1' because it does not exist.
At line:309 char:21
+     $relativePath = Resolve-Path -Relative $path
+                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : ObjectNotFound: (C:\Users\u6\scoop\a…rrent\bin\scoop.ps1:String) [Resolve-Path], ItemNotFoundException
+ FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.ResolvePathCommand
Resolve-Path : Cannot find path 'C:\Users\u6\scoop\apps\scoop\current\bin\scoop.ps1' because it does not exist.
At line:311 char:21
+     $absolutePath = Resolve-Path $path
+                     ~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : ObjectNotFound: (C:\Users\u6\scoop\a…rrent\bin\scoop.ps1:String) [Resolve-Path], ItemNotFoundException
+ FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.ResolvePathCommand
Adding ~\scoop\shims to your path.
Scoop was installed successfully!
Type 'scoop help' for instructions.
```
Running `git config --get-regexp "protocol\..*"` on a system could return something like this:
```log
protocol.http.allow never
protocol.https.allow never
protocol.git.allow never
protocol.file.allow never
protocol.ssh.allow always
```
... or `git config --get protocol.https.allow`:
```log
never
```
... means that HTTPS cannot be used to clone repos.
The actual capability of cloning via HTTPS could be also determined by the git config value `protocol.allow`, if set, or the env variable `GIT_PROTOCOL_FROM_USER` if the value is `user`.
See here: https://git-scm.com/docs/git-config#Documentation/git-config.txt-protocolallow

#### Possible Solution
Instead of trying to parse the git configuration, which could be cumbersome, a check like `!$? -or $LASTEXITCODE -eq 128` right after the `git clone` command should be sufficient.
In case of error you can then proceed with cloning via SSH (`git@github.com:ScoopInstaller/Scoop.git`).

### System details

**Windows version:** 11 (`Microsoft Windows [Version 10.0.22621.1702]`)

**OS architecture:** x64

**PowerShell version:** 7.3.4

**Additional software:** `git version 2.40.1.windows.1`, `OpenSSH_for_Windows_8.6p1, LibreSSL 3.4.3`

#### Scoop Configuration
```json
{
  "last_update": "2023-05-30T15:49:12.9478507+02:00"
}
```

### PR DESCRIPTION
**Pull request by xxthunder**
Reproduced 'git clone failure' as in #57 .

Close #57.

### CODE REVIEW

**Comment by xxthunder** [comment]
@chawyehsu Something like this you had in mind?

File: install.ps1 [hunk 1]
```diff
@@ -575,15 +576,25 @@ function Install-Scoop {
             }
             Write-Verbose "Cloning $SCOOP_PACKAGE_GIT_REPO to $SCOOP_APP_DIR"
             git clone -q $SCOOP_PACKAGE_GIT_REPO $SCOOP_APP_DIR
+            if (-Not $?) {
+                Write-Error "Cloning failed. Falling back to downloading zip files."
```
Comment (chawyehsu) [inline install.ps1:580]: Should `Write-Error` be `throw` and...

File: install.ps1 [hunk 2]
```diff
@@ -575,15 +576,25 @@ function Install-Scoop {
...
+            }
             Write-Verbose "Cloning $SCOOP_MAIN_BUCKET_GIT_REPO to $SCOOP_MAIN_BUCKET_DIR"
             git clone -q $SCOOP_MAIN_BUCKET_GIT_REPO $SCOOP_MAIN_BUCKET_DIR
+            if (-Not $?) {
+                Write-Error "Cloning failed. Falling back to downloading zip files."
+            }
+            $downloadZipsRequired = $False
         } catch {
-            Get-Error $_
+            Write-Output "$($_.Exception.Message)"
```
Comment (chawyehsu) [inline install.ps1:589]: then `Write-Output` be `Write-Warning`.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -1,55 +1,72 @@
...
+                  fetch-depth: 2
+            - name: Test Scoop Install command
+              shell: pwsh
+              run: |
+                  git config --global protocol.https.allow never
+                  ./install.ps1 -RunAsAdmin
+                  echo "~\scoop\shims" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
+            - name: Test scoop command availability
+              shell: pwsh
+              run: scoop help
```
[the remark this answers is not in the document]
Comment (xxthunder) [reply .github/workflows/ci.yml:72]: Sorry for that. Will undo it.

File: .vscode/settings.json [hunk 4]
```diff
@@ -13,5 +13,7 @@
     "files.exclude": {
         "**/.git": true,
         "**/.DS_Store": true
-    }
+    },
+    "iis.configDir": "",
```
[the remark this answers is not in the document]
Comment (xxthunder) [reply .vscode/settings.json:17]: Latest VSCode does this automatically. You are right, we don't need it.

**Comment by xxthunder** [comment]
@chawyehsu Ready for review and merge.

**Review by chawyehsu — APPROVED** [review]
LGTM

**Pull request merged**